### PR TITLE
HRINT-2431 [Demo] Compare Exchange: Create CmpXchg Item & Index CmpXc…

### PR DESCRIPTION
**Pull request by Danielle9897**
…hg Values + Upgrade RavenDB.Client package to 5.3.1

### CODE REVIEW

File: DemoServer/Controllers/Demos/CompareExchange/CreateCompareExchange/CreateCompareExchangeController.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            var getCmpXchgOperation =
+                new GetCompareExchangeValueOperation<string>(cmpXchgKey);
+            
+            CompareExchangeValue<string> getCmpXchgResult =
+                DocumentStoreHolder.Store.Operations.Send(getCmpXchgOperation);
+            #endregion
+            
+            #region Step_4
+            var key = getCmpXchgResult.Key;
+            var value = getCmpXchgResult.Value;
```
Comment (DejanMilicic) [inline DemoServer/Controllers/Demos/CompareExchange/CreateCompareExchange/CreateCompareExchangeController.cs:54]: For some reason, syntax coloring is reacting strange to "value"

image

File: DemoServer/Controllers/Demos/CompareExchange/CreateCompareExchange/metadata.json [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                },
+                {
+                    "Title": "Get Compare-Exchange",
+                    "Type" : "link",
+                    "Url"  : "https://ravendb.net/docs/article-page/5.3/csharp/client-api/operations/compare-exchange/get-compare-exchange-values"
+                }
+            ]
+        },
+        {
+            "Title": "The compare-exchage item",
```
Comment (DejanMilicic) [inline DemoServer/Controllers/Demos/CompareExchange/CreateCompareExchange/metadata.json:143]: typo: "exchage"

File: DemoServer/Controllers/Demos/CompareExchange/IndexCompareExchange/metadata.json [hunk 3]
```diff
@@ -0,0 +1,161 @@
+{
+    "Slug": "index-compare-exchange",
+    "SourceFileName": "IndexCompareExchangeController.cs",
+    "Title": "Index CmpXchg Values",
+    "DescriptionLines": [
+        "**CmpXchg Values can be loaded & indexed** within the index definition.",
+        "",
+        "Any change in the indexed CmpXchg Values will trigger re-indexing."
```
Comment (DejanMilicic) [inline DemoServer/Controllers/Demos/CompareExchange/IndexCompareExchange/metadata.json:8]: any change in the indexed ones, or any change of any cmpXchg value?

File: DemoServer/Controllers/Demos/CompareExchange/IndexCompareExchange/IndexCompareExchangeController.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    public class IndexCompareExchangeController : DemoCodeController
+    {
+        public IndexCompareExchangeController(UserIdContainer userId, UserStoreCache userStoreCache, MediaStoreCache mediaStoreCache,
+            DatabaseSetup databaseSetup) : base(userId, userStoreCache, mediaStoreCache, databaseSetup)
+        {
+        }
+
+        #region Demo
+        #region Step_1
+        public class Products_byUnitsInStock : AbstractIndexCreationTask<Product, Products_byUnitsInStock.IndexEntry>
```
Comment (DejanMilicic) [inline DemoServer/Controllers/Demos/CompareExchange/IndexCompareExchange/IndexCompareExchangeController.cs:26]: is lowercased "by" intentionally lowercased?

File: DemoServer/Controllers/Demos/CompareExchange/IndexCompareExchange/metadata.json [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+            ]
+        },
+        {
+            "Title": "The index Entry class",
+            "Slug": "step-2",
+            "DescriptionLines": [
+                "* The `IndexEntry` class defines the index-fields per entry in the index.<br>",
+                "  These fields will be indexed and can be queried on.",
+                "",
+                "* Field _UnitsInStock_ will be configured to have CmpXchg Values (see next step)."
```
Comment (DejanMilicic) [inline DemoServer/Controllers/Demos/CompareExchange/IndexCompareExchange/metadata.json:80]: maybe "to hold CmpXchg" is slightly more precise wording?

**Review by DejanMilicic — CHANGES_REQUESTED** [review]

File: DemoServer/Controllers/Demos/CompareExchange/CreateCompareExchange/CreateCompareExchangeController.cs (hunk 1 above)
Comment (Danielle9897) [reply DemoServer/Controllers/Demos/CompareExchange/CreateCompareExchange/CreateCompareExchangeController.cs:54]: The color scheme of this editor decides the coloring
So for now I added the prefix '`current`', 
which is not so bad, it reflects that this is the current data we just got...


image

File: DemoServer/Controllers/Demos/CompareExchange/CreateCompareExchange/metadata.json (hunk 2 above)
Comment (Danielle9897) [reply DemoServer/Controllers/Demos/CompareExchange/CreateCompareExchange/metadata.json:143]: fixed

File: DemoServer/Controllers/Demos/CompareExchange/IndexCompareExchange/IndexCompareExchangeController.cs (hunk 4 above)
Comment (Danielle9897) [reply DemoServer/Controllers/Demos/CompareExchange/IndexCompareExchange/IndexCompareExchangeController.cs:26]: fixed to uppercase

File: DemoServer/Controllers/Demos/CompareExchange/IndexCompareExchange/metadata.json (hunk 5 above)
Comment (Danielle9897) [reply DemoServer/Controllers/Demos/CompareExchange/IndexCompareExchange/metadata.json:80]: done

File: DemoServer/Controllers/Demos/CompareExchange/IndexCompareExchange/metadata.json (hunk 3 above)
Comment (Danielle9897) [reply DemoServer/Controllers/Demos/CompareExchange/IndexCompareExchange/metadata.json:8]: as I understand - reindexing will occur for any change made to the indexed cmpxchg values - those that are related/associated with the key that was used in the Map definition

**Review by DejanMilicic — APPROVED** [review]

**Pull request merged**